vc/examples_test.go

```diff
@@ -0,0 +1,150 @@
+package vc_test
+
+import (
+	"encoding/json"
+	"fmt"
+	"time"
+
+	"github.com/tbd54566975/web5-go/dids/didjwk"
+	"github.com/tbd54566975/web5-go/vc"
+)
+
+// Demonstrates how to create, sign, and verify a Verifiable Credential using the vc package.
+func Example() {
+	// create sample issuer and subject DIDs
+	issuer, err := didjwk.Create()
+	if err != nil {
+		panic(err)
+	}
+
+	subject, err := didjwk.Create()
+	if err != nil {
+		panic(err)
+	}
+
+	// creation
+	claims := vc.Claims{"id": subject.URI, "name": "Randy McRando"}
+	cred := vc.Create(claims)
+
+	// signing
+	vcJWT, err := cred.Sign(issuer)
+	if err != nil {
+		panic(err)
+	}
+
+	// verification
+	decoded, err := vc.Verify[vc.Claims](vcJWT)
+	if err != nil {
+		panic(err)
+	}
+
+	fmt.Println(decoded.VC.CredentialSubject["name"])
+	// Output: Randy McRando
+}
+
+type KnownCustomerClaims struct {
+	ID   string `json:"id"`
+	Name string `json:"name"`
+}
+
+func (c KnownCustomerClaims) GetID() string {
+	return c.ID
+}
+
+func (c KnownCustomerClaims) SetID(id string) {
+	c.ID = id
+}
```

Comment:
```suggestion
func (c *KnownCustomerClaims) GetID() string {
	return c.ID
}

func (c *KnownCustomerClaims) SetID(id string) {
	c.ID = id
}
```

And then we'll also have to...

```go
decoded, err := vc.Verify[*KnownCustomerClaims](vcJWT)
```

---

We may think through this a bit more. `vc.Claims` doesn't need to implement pointer receiver methods because maps are reference types (where as structs are value types).

Reply:
i think we have to use a pointer for `SetID()` because otherwise we'd just be updating the copy yeah?

Reply:
that's right, then when we use the type in the generic we have to use a pointer type 

"KnownCustomerClaims does not satisfy vc.CredentialSubject (method SetID has pointer receiver)"

I guess actually we can keep `GetID()` as it is, without a pointer... I had thought we would need to apply it across the entire interface but apparently not